src/compiler/scala/tools/nsc/typechecker/Contexts.scala

```diff
@@ -99,8 +99,79 @@ trait Contexts { self: Analyzer =>
   }
 
 
+  // Collects the leading imports in a given unit body.
+  //
+  // This method should return the list of imports that's
+  // checked in `noPredefImportForUnit`
+  //
+  private[this] def collectLeadingImports(body: Tree): List[Import] = {
+    @tailrec def loop(rem: List[Tree], acc: List[Import]): List[Import] = rem match {
+      case tree :: tail => tree match {
+        case PackageDef(_, stats) => loop(stats ::: tail, acc)
```

Comment:
This drills down through package statements, different from current check.

Reply:
Sorry, I retract that! It does in fact enter nested packages looking for the exclusionary import.